src/components/selection-cards/SelectionCards.tsx

```diff
@@ -0,0 +1,281 @@
+ * It shows a card for each option.
+ *
+ * Example:
+ *
+ * ```
+ *  <SelectionCars
+ *    alignment: GroupAlignment.Horizontal,
+ *    ariaLabel: 'Choose your favorite number',
+ *    onChange: setFavoriteNumber
+ *    options: [
+ *      { label: 'One', value: '1' },
+ *      { label: 'Two', value: '2' },
+ *      { label: 'Three', value: '3' },
+ *    ],
+ *    value: favoriteNumber
+ *  />
+ * ```
+ */
+export const SelectionCards = forwardRef<HTMLDivElement, SelectionCardsProps>((props, ref) => {
+  const {
+    alignment = GroupAlignment.Vertical,
+    ariaLabel,
+    ariaLabelledBy,
+    className,
+    id,
+    isDisabled: disabled,
+    onChange,
+    options,
+    value,
+    ...radixRadioGroupProps
+  } = props;
+
+  const defaultId = `${useId()}selection-cards`;
+
+  return (
+    <SelectionCardsRoot
+      {...radixRadioGroupProps}
+      aria-label={ariaLabel}
+      aria-labelledby={ariaLabelledBy}
+      className={className}
+      data-alignment={alignment}
+      disabled={disabled}
+      id={id ?? defaultId}
+      onValueChange={onChange}
+      ref={ref}
+      value={value}>
+      {options.map(({ isDisabled: disabledOption, ...o }) => (
+        <SelectionCard
+          isDisabled={disabled ? true : disabledOption} // Group disabled takes precedence
+          key={o.value}
+          {...o}
+        />
+      ))}
+    </SelectionCardsRoot>
+  );
+});
+
+SelectionCards.displayName = 'SelectionCards';
+
+function SelectionCard(props: Readonly<SelectionCardOption>) {
+  const { ariaLabel, helpText, illustration, isDisabled, label, value } = props;
+
+  /*
+   * Although the HTML spec defines buttons as valid targets for labels,
+   * some Screen Readers have issues with those:
+   * https://github.com/dequelabs/axe-core/issues/3696#issuecomment-1665575742
+   * So here we force an aria-label:
+   */
+  const inputLabel = ariaLabel ?? (typeof label === 'string' ? label : undefined);
+
+  return (
+    <StyledSelectionCard aria-label={inputLabel} disabled={isDisabled} value={value}>
+      {illustration && <IllustrationContainer>{illustration}</IllustrationContainer>}
+      <SelectionCardContentWrapper hasIllustration={isDefined(illustration)}>
+        <SelectionCardLabel>{label}</SelectionCardLabel>
+        {isDefined(helpText) && (typeof helpText !== 'string' || isStringDefined(helpText)) && (
+          <StyledHelperText {...(isDisabled ? { 'data-disabled': true } : {})}>
+            {helpText}
+          </StyledHelperText>
+        )}
+      </SelectionCardContentWrapper>
+    </StyledSelectionCard>
+  );
+}
+
+SelectionCard.displayName = 'SelectionCard';
+
+const SelectionCardsRoot = styled(RadioGroup.Root)`
+  gap: var(--echoes-dimension-space-200);
+  display: inline-flex;
+  flex-direction: column;
+
+  &[data-alignment='horizontal'] {
+    flex-direction: row;
+  }
+`;
+SelectionCardsRoot.displayName = 'SelectionCardsRoot';
+
+const SelectionCardLabel = styled.label`
```

Comment:
Should we actually use the `Label` component?

Reply:
Yeah I think we should, I just pinged Marcio about that: 
![image](https://github.com/user-attachments/assets/9c8682cc-239d-4fbb-8e65-07483ce2a9a4)